janus.doc

```diff
@@ -152,8 +152,8 @@ py_set(List) & $\Longleftrightarrow$ & Set & \\
 -()     & $\Longleftrightarrow$ & ()  & Python empty Tuple \\
 -(a,b,\ldots) & $\Longleftrightarrow$ & (a,b,\ldots) & Python Tuples.  Note that a Prolog \jargon{pair} \exam{A-B} maps to a Python (binary) tuple. \\
 Dict    & $\Longleftrightarrow$ & Dict     & Default for SWI-Prolog \\
-\{k:v,\ldots\} & $\Longleftrightarrow$ & Dict & Compatibility when using \exam{py_dict_as({{}})}\\
-py(\{\mbox{}\}) & $\longleftarrow$ & \{\mbox{}\} & Empty dict when using \exam{py_dict_as({{}})}\\
```

Comment:
That should not be needed.    It translates (see build/packages/swipy/janus.tex) into ``py_dict_as({\Scurl})``, which renders correctly as {...} merely defines a group in LaTeX.    Looks fine in both the web page and PDF doc.

The idea is that {Special} is translated into `{\Command}`, where `\Command` is defined to produce the desired output in a way that can be embedded anywhere.    This allows passing these hard to manage sequences to commands anywhere.    If you need to write the sequence in running text, you use {Special} to trigger the translation to `{\Command}`.

Reply:
There appears to be something wrong when I do a build of `packages/swipy`, and the transformation doesn't happen. (Which is why I made  the change in `janus.doc`.
One clue: when I do a debug build, there's nothing under `packages/swipy` except for the cmake files - all other packages appear to build correctly for debug.

I didn't understand your explanation of `{Special}`. (That's one reason I was trying to generate the `ltx2htm` documentation)

Reply:
The debug build for Prolog tries to find the debug version of Python.    I think it should fall back to the normal version, but probably that doesn't work as expected.    Either see what is going wrong in this Cmake bit or, more easily, install the debug version of libpython.

The "Special" hacking requires quite a bit of understanding of LaTeX.   I think you lack that background knowledge, no?     These are work arounds to make it fairly convenient to deal with Prolog atoms such as `>=` or `\+`.   By writing them as {\+}, the pre-processing identifies them as special and avoids the need for very complicated escape sequences that vary depending on the context in which you want to use these symbols.    LaTeX reserves a lot of stuff to make math typesetting comfortable, but this interferes with documenting text about programming :cry: